src/EFCore.Relational/Query/SqlExpressionFactory.cs

```diff
@@ -814,7 +814,7 @@ private void AddConditions(SelectExpression selectExpression, IEntityType entity
         {
             AddSelfConditions(selectExpression, entityType, table);
             // Add inner join to principal if table sharing
-            table ??= entityType.GetViewOrTableMappings().SingleOrDefault()?.Table;
+            table ??= entityType.GetRootType().GetViewOrTableMappings().SingleOrDefault()?.Table;
```

Comment:
Should return the same result, but less code
```suggestion
            table ??= entityType.GetRootType().GetViewOrTableMappings().FirstOrDefault()?.Table;
```